### DISCUSSION TITLE
thread/ReentrantReadWriteLock

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#293] [Announcements]
# thread/ReentrantReadWriteLock

ReentrantReadWriteLock 是 Java 的一种读写锁，它允许多个读线程同时访问，但只允许一个写线程访问，或者阻塞所有的读写线程。这种锁的设计可以提高性能，特别是在读操作的数量远远超过写操作的情况下。

https://javabetter.cn/thread/ReentrantReadWriteLock.html

**Comment by Ricehahaa** [comment]
看源码就是蓝受

**Comment by LFAW233** [comment]
怎么有些源码对齐会变得稀烂啊，如锁降级中的，缩紧没有一个是对的上的
